File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        st=s[len(s)-1: : -1]
        list=st.split()
        s,e=0,len(list)-1
        while(s<e):
            temp=list[s]
            list[s]=list[e]
            list[e]=temp
            s+=1
            e-=1
        ans=' '.join(list)
        return ans

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        String arr[]=s.split(" ");
        String str="";
        for(int i=0; i<arr.length; i++){
            String st=arr[i];
            for(int j=st.length()-1; j>=0; j--){
                str+=st.charAt(j);
            }
            if(i!=arr.length-1)str+=' ';
        }
        return str;
    }
}